File: XUCore/additions/NSOpenPanelAdditions.m
// 
// NSOpenPanelAdditions.m
// 
// Created by Charlie Monroe
// 
// Copyright (c) 2010-13 Fuel Collective, LLC. All rights reserved.
// 


#import "NSOpenPanelAdditions.h"

@implementation NSOpenPanel (NSOpenPanelAdditions)

-(NSInteger)runModalOnMainThread{
	if ([NSThread currentThread] == [NSThread mainThread]){
		return [self runModal];
	}
	
	NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:[self methodSignatureForSelector:@selector(runModal)]];
	[invocation setSelector:@selector(runModal)];
	[invocation performSelectorOnMainThread:@selector(invokeWithTarget:) withObject:self waitUntilDone:YES];
	
	void *result = NULL;
	[invocation getReturnValue:&result];
	
	return (NSInteger)result;
}

@end

File: XUCore/additions/NSOpenPanelAdditions.swift
//
//  NSOpenPanelAdditions.swift
//  XUCore
//
//  Created by Charlie Monroe on 11/8/15.
//  Copyright © 2015 Charlie Monroe Software. All rights reserved.
//

import Foundation

public extension NSOpenPanel {
	
	public func runModalOnMainThread() -> Int {
		var response = NSFileHandlingPanelCancelButton
		XU_PERFORM_BLOCK_ON_MAIN_THREAD { () -> Void in
			response = self.runModal()
		}
		
		return response
	}
	
}
